diff --git a/libavcodec/encode.c b/libavcodec/encode.c
index 407bd89..ebfbd91 100644
--- a/libavcodec/encode.c
+++ b/libavcodec/encode.c
@@ -768,7 +768,6 @@ int ff_encode_preinit(AVCodecContext *avctx)
     }
 
     switch (avctx->codec_type) {
-    case AVMEDIA_TYPE_VIDEO: ret = encode_preinit_video(avctx); break;
     case AVMEDIA_TYPE_AUDIO: ret = encode_preinit_audio(avctx); break;
     }
     if (ret < 0)
diff --git a/libavcodec/get_buffer.c b/libavcodec/get_buffer.c
index b391adf..49563bb 100644
--- a/libavcodec/get_buffer.c
+++ b/libavcodec/get_buffer.c
@@ -82,59 +82,6 @@ static int update_frame_pool(AVCodecContext *avctx, AVFrame *frame)
         return AVERROR(ENOMEM);
 
     switch (avctx->codec_type) {
-    case AVMEDIA_TYPE_VIDEO: {
-        int linesize[4];
-        int w = frame->width;
-        int h = frame->height;
-        int unaligned;
-        ptrdiff_t linesize1[4];
-        size_t size[4];
-
-        avcodec_align_dimensions2(avctx, &w, &h, pool->stride_align);
-
-        do {
-            // NOTE: do not align linesizes individually, this breaks e.g. assumptions
-            // that linesize[0] == 2*linesize[1] in the MPEG-encoder for 4:2:2
-            ret = av_image_fill_linesizes(linesize, avctx->pix_fmt, w);
-            if (ret < 0)
-                goto fail;
-            // increase alignment of w for next try (rhs gives the lowest bit set in w)
-            w += w & ~(w - 1);
-
-            unaligned = 0;
-            for (i = 0; i < 4; i++)
-                unaligned |= linesize[i] % pool->stride_align[i];
-        } while (unaligned);
-
-        for (i = 0; i < 4; i++)
-            linesize1[i] = linesize[i];
-        ret = av_image_fill_plane_sizes(size, avctx->pix_fmt, h, linesize1);
-        if (ret < 0)
-            goto fail;
-
-        for (i = 0; i < 4; i++) {
-            pool->linesize[i] = linesize[i];
-            if (size[i]) {
-                if (size[i] > INT_MAX - (16 + STRIDE_ALIGN - 1)) {
-                    ret = AVERROR(EINVAL);
-                    goto fail;
-                }
-                pool->pools[i] = av_buffer_pool_init(size[i] + 16 + STRIDE_ALIGN - 1,
-                                                     CONFIG_MEMORY_POISONING ?
-                                                        NULL :
-                                                        av_buffer_allocz);
-                if (!pool->pools[i]) {
-                    ret = AVERROR(ENOMEM);
-                    goto fail;
-                }
-            }
-        }
-        pool->format = frame->format;
-        pool->width  = frame->width;
-        pool->height = frame->height;
-
-        break;
-        }
     case AVMEDIA_TYPE_AUDIO: {
         ret = av_samples_get_buffer_size(&pool->linesize[0],
                                          frame->ch_layout.nb_channels,
@@ -254,35 +201,10 @@ int avcodec_default_get_buffer2(AVCodecContext *avctx, AVFrame *frame, int flags
 {
     int ret;
 
-    if (avctx->hw_frames_ctx) {
-        ret = av_hwframe_get_buffer(avctx->hw_frames_ctx, frame, 0);
-        if (ret == AVERROR(ENOMEM)) {
-            AVHWFramesContext *frames_ctx =
-                (AVHWFramesContext*)avctx->hw_frames_ctx->data;
-            if (frames_ctx->initial_pool_size > 0 &&
-                !avctx->internal->warned_on_failed_allocation_from_fixed_pool) {
-                av_log(avctx, AV_LOG_WARNING, "Failed to allocate a %s/%s "
-                       "frame from a fixed pool of hardware frames.\n",
-                       av_get_pix_fmt_name(frames_ctx->format),
-                       av_get_pix_fmt_name(frames_ctx->sw_format));
-                av_log(avctx, AV_LOG_WARNING, "Consider setting "
-                       "extra_hw_frames to a larger value "
-                       "(currently set to %d, giving a pool size of %d).\n",
-                       avctx->extra_hw_frames, frames_ctx->initial_pool_size);
-                avctx->internal->warned_on_failed_allocation_from_fixed_pool = 1;
-            }
-        }
-        frame->width  = avctx->coded_width;
-        frame->height = avctx->coded_height;
-        return ret;
-    }
-
     if ((ret = update_frame_pool(avctx, frame)) < 0)
         return ret;
 
     switch (avctx->codec_type) {
-    case AVMEDIA_TYPE_VIDEO:
-        return video_get_buffer(avctx, frame);
     case AVMEDIA_TYPE_AUDIO:
         return audio_get_buffer(avctx, frame);
     default:
diff --git a/libavutil/frame.c b/libavutil/frame.c
index be30eb0..a82608c 100644
--- a/libavutil/frame.c
+++ b/libavutil/frame.c
@@ -208,8 +208,6 @@ int av_frame_get_buffer(AVFrame *frame, int align)
     if (frame->format < 0)
         return AVERROR(EINVAL);
 
-    if (frame->width > 0 && frame->height > 0)
-        return get_video_buffer(frame, align);
     else if (frame->nb_samples > 0 &&
              (av_channel_layout_check(&frame->ch_layout)))
         return get_audio_buffer(frame, align);
@@ -713,8 +711,6 @@ int av_frame_copy(AVFrame *dst, const AVFrame *src)
     if (dst->format != src->format || dst->format < 0)
         return AVERROR(EINVAL);
 
-    if (dst->width > 0 && dst->height > 0)
-        return frame_copy_video(dst, src);
     else if (dst->nb_samples > 0 &&
              (av_channel_layout_check(&dst->ch_layout)))
         return frame_copy_audio(dst, src);
